src/app/ui/earn/utils.ts

```diff
@@ -0,0 +1,130 @@
+import { map, uniqBy, uniq, sortBy, fromPairs } from 'lodash';
+import { EarnCardVariant } from 'src/components/Cards/EarnCard/EarnCard.types';
+import { EarnOpportunityWithLatestAnalytics } from 'src/types/jumper-backend';
+import {
+  EarnFilteringParams,
+  EarnsPageSearchParams,
+  FilterKey,
+  SortByOptions,
+} from './types';
+import { EarnOpportunityFilter } from 'src/app/lib/getOpportunitiesFiltered';
+import { toFixedFractionDigits } from 'src/utils/formatNumbers';
+
+const parseString = (value: string): string | null => {
```

Comment:
Instead of rolling our own serialize / deserialize, could we use something like nuqs? It seems to be well maintained and is nextjs first (I guess we can easily use it in ssr and all).

https://www.npmjs.com/package/nuqs